grabbit/src/main/groovy/com/twcable/grabbit/spring/batch/repository/GrabbitJobExecutionDao.groovy

```diff
@@ -0,0 +1,23 @@
+package com.twcable.grabbit.spring.batch.repository
+
+import org.springframework.batch.core.BatchStatus
+import org.springframework.batch.core.JobExecution
+import org.springframework.batch.core.repository.dao.JobExecutionDao
+
+/**
+ * Modified DAO Interface for persisting and retrieving {@link JobExecution}
+ * @see JobExecutionDao for more details
+ */
+interface GrabbitJobExecutionDao extends JobExecutionDao{
+
+    /**
+     * Returns job execution paths for given BatchStatuses
+     */
+    public List<String> getJobExecutions(List<BatchStatus> batchStatuses)
```

Comment:
Just to be clear, Jeff was channelling me and my insistence on the semantics of collection types. `List` implies an importance to the order. If ordering does-not/should-not matter, is should be a `Collection` or `Set`. !fix